code/main.cpp

```diff
@@ -1,24 +1,24 @@
 #include <hwlib.hpp>
 #include <i2c_bus.hpp>
-#include <ssd1306_oled_unbuffered.hpp>
+#include <ssd1306_oled_buffered.hpp>
 int main(void) {
     // kill the watchdog
     WDT->WDT_MR = WDT_MR_WDDIS;
     // this starts the sysclock
     hwlib::wait_ms(20);
 
-    hwlib::cout << "Before constructor" << hwlib::endl;
     auto bus = r2d2::i2c::i2c_bus_c(
-        r2d2::i2c::i2c_bus_c::interface::interface_0, 50000);
-    hwlib::cout << "After constructor" << hwlib::endl;
-    auto display = r2d2::display::ssd1306_oled_unbuffered(bus, 0x3c);
-    hwlib::cout << "after r2d2 display" << hwlib::endl;
+        r2d2::i2c::i2c_bus_c::interface::interface_0,  400000);
+    auto display = r2d2::display::ssd1306_oled_buffered(bus, 0x3c);
     display.clear();
-    hwlib::cout << "After clear" << hwlib::endl;
+    display.flush();
```

Comment:
Maybe add some comments to the main if someone wants to read through it without knowledge.

Reply:
Added comments to the main, will also add doxygen.